controllers/btpoperator_controller.go

```diff
@@ -1131,3 +1201,480 @@ func (r *BtpOperatorReconciler) watchConfigPredicates() predicate.Funcs {
 		UpdateFunc: func(e event.UpdateEvent) bool { return nameMatches(e.ObjectNew) },
 	}
 }
+
+func (r *BtpOperatorReconciler) watchWebhooksPredicates() predicate.Funcs {
+	predicateForCertificates := func(obj client.Object) bool {
+		switch v := obj.(type) {
+		case *admissionregistrationv1.ValidatingWebhookConfiguration:
+			{
+				return v.Name == validatingWebhookName
+			}
+		case *admissionregistrationv1.MutatingWebhookConfiguration:
+			{
+				return v.Name == mutatingWebhookName
+			}
+		default:
+			return false
+		}
+	}
+
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return predicateForCertificates(e.Object)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return predicateForCertificates(e.Object)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return predicateForCertificates(e.ObjectOld)
+		},
+	}
+}
+
+func (r *BtpOperatorReconciler) reconcileCertificates(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured) error {
```

Comment:
This is very long function. Consider separating some parts like:
- secret existence validation
- certs validation